test/jdk/javax/swing/text/html/HtmlTagParserTest.java

```diff
@@ -0,0 +1,105 @@
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8012675
+ * @library /java/awt/regtesthelpers
+ * @build PassFailJFrame
+ * @summary Tests if HTML script tags are unsupported
+ * @run main/manual HtmlTagParserTest
+ */
+
+import java.awt.Dimension;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.lang.reflect.InvocationTargetException;
+import java.net.URL;
+import java.nio.file.Path;
+
+import javax.swing.JEditorPane;
+import javax.swing.JFrame;
+import javax.swing.JScrollPane;
+import javax.swing.SwingUtilities;
+
+public class HtmlTagParserTest {
+    private static String instructionsText = "Pass if you can see the " +
+            "script tag and its contents as text in JTextFields and " +
+            "JTextAreas in the frame. Fail if this is not shown.";
+
+    private static final String htmlText = "<!DOCTYPE html> <html> <head> " +
+            "<title>Title</title> </head> <body><script>function foo() " +
+            "{alert('Alert generated by script tag function!');}</script>" +
+            "This is a test. <input onclick=\"foo()\" type=\"button\" " +
+            "value=\"Click Me\" id=\"myButton1\"></input> </body> </html>";
+
+    private static JFrame frame;
+    private static JEditorPane jep;
+    private static Path testDir;
+
+    public static void createAndShowGUI() throws InterruptedException,
+            InvocationTargetException {
+        SwingUtilities.invokeAndWait(() -> {
+            jep = new JEditorPane();
+            JScrollPane scroll = new JScrollPane(jep);
+            Dimension scrollSize = new Dimension(500, 300);
+            scroll.setPreferredSize(scrollSize);
+            scroll.setVerticalScrollBarPolicy(JScrollPane.VERTICAL_SCROLLBAR_ALWAYS);
+
+            frame = new JFrame();
+            frame.getContentPane().add(scroll);
+            frame.setVisible(true);
+            frame.pack();
+            frame.setLocationRelativeTo(null);
+
+            PassFailJFrame.addTestFrame(frame);
+            PassFailJFrame.positionTestFrame(frame,
+                    PassFailJFrame.Position.HORIZONTAL);
+        });
+    }
+
+    public static void main(String args[]) throws InterruptedException,
```

Comment:
Thanks for catching that. Also, the reasoning for the test originally using a temporary file to create HTML was to have an HTML file that could be opened itself and compared towards. I see that it's not necessary for this test, but was useful at the time. I thought it could still be useful, but this test with the HTML stored in a string is probably sufficient.

Reply:
> the reasoning for the test originally using a temporary file to create HTML was to have an HTML file that could be opened itself and compared towards. I see that it's not necessary for this test, but was useful at the time. I thought it could still be useful, but this test with the HTML stored in a string is probably sufficient.

It could still be useful. An option to the test, say `-saveHTML`, could be used to get the HTML file to open in browser.

Yet I think it's easier to run the test on its own when you're developing it, no need to define the system property.

Reply:
I can look into this as well. Does this exception occur with the current version without the HTML file? I have no issues on my end.

Reply:
I'll remove the button then as it's not necessary for the test. I mainly added it to demonstrate that HTML contents are rendering correctly. I'll make this change and update the doc for Parser to more explicitly explain what happens when its editable vs non-editable.

Reply:
It looks good overall.

It started with `<script>` tag; the text in javadoc refers to other unknown or unsupported tags. Shall the test be expanded with an unknown tag?

Displaying two instances of `JEditorPane` where one's editable and the other isn't could be too confusing. Only editable is enough, I guess.

Reply:
Sure, I added a foo tag with text inside. Behavior still displays as expected and stated in the doc.